src/main/java/org/fogbowcloud/manager/core/plugins/identity/ldap/LdapIdentityPlugin.java

```diff
@@ -0,0 +1,344 @@
+import java.util.Hashtable;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+import javax.naming.Context;
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.InitialDirContext;
+import javax.naming.directory.SearchControls;
+import javax.naming.directory.SearchResult;
+
+import org.apache.commons.codec.Charsets;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.http.HttpStatus;
+import org.apache.log4j.Logger;
+import org.fogbowcloud.manager.core.models.Credential;
+import org.fogbowcloud.manager.core.models.token.Token;
+import org.fogbowcloud.manager.core.plugins.IdentityPlugin;
+import org.fogbowcloud.manager.core.plugins.identity.util.RSAUtils;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class LdapIdentityPlugin implements IdentityPlugin {
+
+	private static final String ATT_EXPIRATION_DATE = "expirationDate";
+	private static final String ATT_NAME = "name";
+	private static final String ATT_LOGIN = "lognin";
+
+	private static final Logger LOGGER = Logger.getLogger(LdapIdentityPlugin.class);
+	private static final long EXPIRATION_INTERVAL = TimeUnit.DAYS.toMillis(365); // One
+																					// year
+
+	private static final String PROP_LDAP_BASE = "ldap_base";
+	private static final String PROP_LDAP_URL = "ldap_identity_url";
+	private static final String PROP_LDAP_ENCRYPT_TYPE = "ldap_encrypt_type";
+
+	private static final String PRIVATE_KEY_PATH = "private_key_path";
+	private static final String PUBLIC_KEY_PATH = "public_key_path";
+	
+	public static final String CRED_USERNAME = "username";
+	public static final String CRED_PASSWORD = "password";
+	public static final String CRED_AUTH_URL = "authUrl";
+	public static final String CRED_LDAP_BASE = "base";
+	public static final String CRED_LDAP_ENCRYPT = "encrypt";
+	public static final String CRED_PRIVATE_KEY = "privateKey";
+	public static final String CRED_PUBLIC_KEY = "publicKey";
+	public static final String ENCRYPT_TYPE = ":TYPE:";
+	public static final String ENCRYPT_PASS = ":PASS:";
+	public static final String PASSWORD_ENCRYPTED = "{" + ENCRYPT_TYPE + "}" + ENCRYPT_PASS;
+	public static final String ACCESSID_SEPARATOR = "!#!";
+
+	private String ldapBase;
+	private String ldapUrl;
+	private String encryptType;
+
+	private String privateKeyPath;
+	private String publicKeyPath;
+
+	public LdapIdentityPlugin(Properties properties) {
+		this.ldapBase = properties.getProperty(PROP_LDAP_BASE);
+		this.ldapUrl = properties.getProperty(PROP_LDAP_URL);
+		this.encryptType = properties.getProperty(PROP_LDAP_ENCRYPT_TYPE);
+		this.privateKeyPath = properties.getProperty(PRIVATE_KEY_PATH);
+		this.publicKeyPath = properties.getProperty(PUBLIC_KEY_PATH);
+	}
+
+	@Override
+	public Token createToken(Map<String, String> userCredentials) {
+
+		String uid = userCredentials.get(CRED_USERNAME);
```

Comment:
Change to `userId` or `uuid`.

Reply:
Done :). Check [this](https://github.com/fogbow/fogbow-manager-core/commit/78864ce8eefde35b7118f3caa28d205a7ff8ce8d) out.